tests/integration/operator/operator_test.go

```diff
@@ -0,0 +1,200 @@
+					Success("Cleanup done")
+				})
+
+				When("the resource is created", func() {
+					It("updates the Istio resource status to Reconcilied and Ready", func() {
+						Eventually(kubectl.GetResourceCondition).
+							WithArguments(controlPlaneNamespace, "istio", istioName).
+							Should(ContainElement(resourceReconcilied), "Istio it's not Reconcilied; unexpected Condition")
+
+						Eventually(kubectl.GetResourceCondition).
+							WithArguments(controlPlaneNamespace, "istio", istioName).
+							Should(ContainElement(resourceReady), "Istio it's not Ready; unexpected Condition")
+
+						Eventually(kubectl.GetPodPhase).
+							WithArguments(controlPlaneNamespace, "app=istiod").
+							Should(Equal("Running"), "Istiod is not running; unexpected pod Phase")
+						Success("Istio resource is Reconcilied and Ready")
+					})
+
+					It("deploys correct istiod image tag according to the version in the Istio CR", func() {
+						// TODO: we need to add a function to get the istio version from the control panel directly
+						// and compare it with the applied version
+						// This is a TODO because actual version.yaml contains for example latest and not the actual version
+						// Posible solution is to add actual version to the version.yaml
+					})
+
+					It("doesn't continuously reconcile the istio resource", func() {
+						Eventually(kubectl.GetPodLogs).
+							WithArguments(controlPlaneNamespace, "app=istiod", 30*time.Second).
+							ShouldNot(ContainSubstring("Reconciliation done"), "Istio Operator is continuously reconciling")
+						Success("Istio Operator stopped reconciling")
+					})
+
+					It("deploys istiod", func() {
+						Expect(kubectl.GetDeploymentsNames(controlPlaneNamespace)).
+							To(Equal([]string{"istiod"}), "Istiod deployment is not present; expected list to be equal")
+						Success("Istiod deployment is present")
+					})
+
+					It("deploys the CNI DaemonSet when running on OpenShift", func() {
+						if ocp == "true" {
+							Eventually(kubectl.GetDaemonSets).
+								WithArguments(namespace).
+								Should(ContainElement("istio-cni-node"), "CNI DaemonSet is not deployed; expected list to contain element")
+
+							Expect(kubectl.GetPodPhase(namespace, "k8s-app=istio-cni-node")).Should(Equal("Running"), "CNI Daemon is not running; unexpected Phase")
+							Success("CNI DaemonSet is deployed in the namespace and Running")
+						} else {
+							Consistently(kubectl.GetDaemonSets).
+								WithArguments(namespace).WithTimeout(30*time.Second).
+								Should(BeEmpty(), "CNI DaemonSet it's present; expected list to be empty")
+							Success("CNI DaemonSet is not deployed in the namespace because it's not OpenShift")
+						}
+					})
+				})
+
+				When("the Istio CR is deleted", func() {
+					BeforeEach(func() {
+						Expect(deleteIstioCR(version)).Should(Succeed(), "Istio CR deletion failed; unexpected error")
```

Comment:
We can probably remove the string "unexpected error" here, as it will be clear from the output.